go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -215,6 +215,45 @@ func (root *RootValue) GenerateTagsForNewColumns(ctx context.Context, tableName
 		})
 	}
 
+	if headRoot != nil {
```

Comment:
Move this whole block above the rest of the logic, it's a totally separate code path